prime-router/src/test/kotlin/metadata/MapperTests.kt

```diff
@@ -80,6 +82,178 @@ class MapperTests {
         assertThat(mapper.apply(lookupElement, args, elementAndValues).value).isEqualTo("y")
     }
 
+    private fun assertEq(
+        mapper: Mapper,
+        elm: Element,
+        args: List<String>,
+        EAVs: List<ElementAndValue>,
+        compVal: String
+    ) {
+        assertThat(
+            mapper.apply(elm, args, EAVs).value
+        ).isEqualTo(compVal)
+    }
+
+    @Test
+    fun `test IfThenElseMapper`() {
+        val mapper = IfThenElseMapper()
+        val element = Element("test")
+        var args = mutableListOf<String>()
+        while (args.count() < 11) { // test for zero to ten args passed - each should fail (skip testing 5)
+            if (args.count() != 5) assertThat { mapper.valueNames(element, args) }.isFailure()
+            args.add("arg")
+        }
+        // test normal call  mapper: ifThenElse(<=, otc_flag comparisonValue, patient_state, ordering_provider_state)
+        args = mutableListOf("<=", "otc_flag", "comparisonValue", "patient_state", "ordering_provider_state")
+        val valNames = mapper.valueNames(element, args)
+        assertThat(valNames.count()).isEqualTo(5)
+        assertThat(valNames[0]).isEqualTo("<=")
+        assertThat(valNames[1]).isEqualTo("otc_flag")
+        assertThat(valNames[2]).isEqualTo("comparisonValue")
+        assertThat(valNames[3]).isEqualTo("patient_state")
+        assertThat(valNames[4]).isEqualTo("ordering_provider_state")
+
+        // In all these cases, think of "AL" as the "that's right" and "TN" as the "else case"
+        val eAVotc = ElementAndValue(Element(args[1]), "OTC")
+        val eAVpresc = ElementAndValue(Element(args[2]), "Prescription")
+        val eAValabama = ElementAndValue(Element(args[3]), "AL")
+        val eAVtn = ElementAndValue(Element(args[4]), "TN")
+
+        // test equality else    ("OTC" != "Prescription")
```

Comment:
Did not understand this comment

Reply:
// test 'else value' for ==    ("OTC" != "Prescription")